lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -1684,6 +1684,18 @@ static bool foldMuxChain(MuxOp rootMux, bool isFalseSide,
 LogicalResult MuxOp::canonicalize(MuxOp op, PatternRewriter &rewriter) {
   APInt value;
 
+  // mux(a, 0, 1) -> ~a for single-bit values.
+  if (matchPattern(op.trueValue(), m_RConstant(value)) &&
+      value.getBitWidth() == 1 && value.isZero()) {
+    APInt value2;
+    if (matchPattern(op.falseValue(), m_RConstant(value2)) &&
+        value2.isAllOnes()) {
+      auto one = rewriter.create<hw::ConstantOp>(op.getLoc(), APInt(1, 1));
+      rewriter.replaceOpWithNewOp<XorOp>(op, op.cond(), one);
+      return success();
```

Comment:
This logic LGTM, but please merge it into the stuff below so the "trueValue" is only matched to a constant once.  One additional tweak: can you reuse `op.trueValue()` (which you known is a single bit one value) rather than creating a new "one" constantop?

Reply:
Sounds good, and good call on reusing `op.trueValue()`.